app-backend/tile/src/main/scala/LayerCache.scala

```diff
@@ -53,39 +53,56 @@ object LayerCache extends Config with LazyLogging with KamonTrace {
   private val tileCache = HeapBackedMemcachedClient(memcachedClient)
   private val astCache = HeapBackedMemcachedClient(memcachedClient)
 
-  private val attributeStoreCache: ScaffeineCache[UUID, OptionT[Future, (AttributeStore, Map[String, Int])]] =
+  private val attributeStoreWithMaxZoomsCache: ScaffeineCache[UUID, OptionT[Future, (AttributeStore, Map[String, Int])]] =
     Scaffeine()
       .recordStats()
       .expireAfterAccess(5.minutes)
       .maximumSize(500)
       .build[UUID, OptionT[Future, (AttributeStore, Map[String, Int])]]
 
-  def attributeStoreForLayer(layerId: UUID)(implicit ec: ExecutionContext, projectLayerIds: Set[UUID]): OptionT[Future, (AttributeStore, Map[String, Int])] =
+  private val attributeStoreCache: ScaffeineCache[UUID, OptionT[Future, AttributeStore]] =
+    Scaffeine()
+      .recordStats()
+      .expireAfterAccess(5.minutes)
+      .maximumSize(500)
+      .build[UUID, OptionT[Future, AttributeStore]]
+
+  // TODO: consider having one attribute store per project, layerId should be removed from here
+  def attributeStoreForLayerWithMaxZooms(layerId: UUID)(implicit ec: ExecutionContext, projectLayerIds: Set[UUID]): OptionT[Future, (AttributeStore, Map[String, Int])] =
+    traceName(s"LayerCache.attributeStoreForLayerWithMaxZooms($layerId)") {
+      attributeStoreWithMaxZoomsCache.take(layerId, _ =>
+        traceName(s"LayerCache.attributeStoreForLayerWithMaxZooms($layerId) (no cache)") {
+          val store = PostgresAttributeStore()
+          val maxZooms: Map[String, Int] = store.maxZoomsForLayers(projectLayerIds.map(_.toString))
+          OptionT.fromOption((store, maxZooms).some)
+        }
+      )
+    }
+
+  def attributeStoreForLayer(layerId: UUID)(implicit ec: ExecutionContext): OptionT[Future, AttributeStore] =
     traceName(s"LayerCache.attributeStoreForLayer($layerId)") {
       attributeStoreCache.take(layerId, _ =>
         traceName(s"LayerCache.attributeStoreForLayer($layerId) (no cache)") {
-          val store = PostgresAttributeStore()
-          val maxZooms: Map[String, Int] = blocking { store.maxZoomsForLayers(projectLayerIds.map(_.toString)) }
-          OptionT.fromOption((store, maxZooms).some)
+          OptionT.fromOption(PostgresAttributeStore().some)
```

Comment:
`attributeStoreForLayerWithMaxZooms` function is required only for `Mosaic` operations. I wanted to remove a lot of `implicit sceneIds` params from our api as it's useless. 

I decided to keep `attributeStoreForLayer` _logic_ as it was before, but i think it's a good note, that it should not require caching. 

It can be even a single `val` and not a function (for everything we can use the same attribute store), but it would require some code change and we can do it in a separate PR or i can do it in additional commit.